### PR TITLE
Bind to curl_multi_wait instead of curl_multi_fdset + select

### PR DESCRIPTION
**Pull request by nojb**
Hello,

I am trying to use `Curl.Multi.{perform,wait}` under Windows. Unfortunately it does not really work because `curl_mult_fdset` returns `-1` in the `maxfd` argument, signaling that `select` cannot be used (see doc at https://curl.haxx.se/libcurl/c/curl_multi_fdset.html).

Instead I tried with the function `curl_multi_wait` which is the recommended way to do select-style polling (see doc at https://curl.haxx.se/libcurl/c/curl_multi_wait.html). This seems to work fine and the C code is simpler.

This PR replaces the use of `curl_mult_fdset` + `select` by the simpler use of `curl_multi_wait` which has the advantage of working under Windows and also (according to the documentation) behaves better with respect to the usual limit of file descriptors of `select`.

### CODE REVIEW

File: curl-helper.c [hunk 1]
```diff
@@ -3995,46 +3995,21 @@ value caml_curlm_remove_finished(value v_multi)
...
-  ret = curlm_wait_data(h);
+  ret = curl_multi_wait(multi_handle, NULL, 0, 1000, &numfds);
+  if (ret != CURLM_OK) {
+    caml_leave_blocking_section();
+    caml_failwith("caml_curl_multi_wait");
+  }
   caml_leave_blocking_section();
-
-  CAMLreturn(Val_bool(0 == ret));
+  CAMLreturn(Val_bool(numfds == 0));
```
Comment (ygrek) [inline curl-helper.c]: condition seems backwards : 
```
  (** wait till there are some active data transfers on multi stack
      @return whether [perform] should be called *)
```
ie should return true, when there are some fds with activity.

File: curl-helper.c [hunk 2]
```diff
@@ -3995,46 +3995,21 @@ value caml_curlm_remove_finished(value v_multi)
...
   CAMLparam1(v_multi);
-  int ret;
-  CURLM* h = CURLM_val(v_multi);
+  CURLM *multi_handle = CURLM_val(v_multi);
+  int numfds = -1;
+  CURLMcode ret;
 
   caml_enter_blocking_section();
-  ret = curlm_wait_data(h);
+  ret = curl_multi_wait(multi_handle, NULL, 0, 1000, &numfds);
```
Comment (ygrek) [inline curl-helper.c]: lets allow passing this timeout, as optional integer timeout_ms parameter

**Comment by nojb** [comment]
Thanks for the review! I think I addressed both points. I left the default value for `timeout_ms` set to `1000`.

Also, I am not sure if the build system needs to be updated in any way (to test availability of `curl_multi_wait` or similar).

File: curl-helper.c [hunk 3]
```diff
@@ -3995,15 +3995,19 @@ value caml_curlm_remove_finished(value v_multi)
...
+value caml_curl_multi_wait(value v_timeout_ms, value v_multi)
 {
   CAMLparam1(v_multi);
   CURLM *multi_handle = CURLM_val(v_multi);
+  int timeout_ms = 1000;
   int numfds = -1;
   CURLMcode ret;
 
+  if (v_timeout_ms != Val_int(0))
+    timeout_ms = Int_val(Field(0, v_timeout_ms));
```
Comment (ygrek) [inline curl-helper.c]: %) this is a crash. Better do it on ocaml side

**Comment by ygrek** [comment]
re build system - libcurl version check is enough. Thanks for the PR!

File: curl-helper.c (hunk 3 above)
Comment (nojb) [reply curl-helper.c]: Oops, indeed!

**Pull request merged**